### PR TITLE
Add code comments to sstable

### PR DESCRIPTION
**Pull request by naveen246**

### CODE REVIEW

**Comment by naveen246** [comment]
@purplefox 
Added the code comments since I felt it helps in understanding the code.
Used https://github.com/ArthurSonzogni/Diagon to generate the ascii table

File: sst/sstable.go [hunk 1]
```diff
@@ -20,9 +20,34 @@ type SSTable struct {
...
+	//  │format│metadataOffset│
+	//  ├──────┼──────────────┤
+	//  │1 byte│ 4 bytes      │
+	//  ╰──────┴──────────────╯
+	//  ╭─────────┬────────────────┬────────────┬──────────────────╮
+	//  │keyLength│ key            │ valueLength│ value            │
+	//  ├─────────┼────────────────┼────────────┼──────────────────┤ ... repeat KV pairs
+	//  │4 bytes  │ keyLength bytes│ 4 bytes    │ valueLength bytes│
+	//  ╰─────────┴────────────────┴────────────┴──────────────────╯
+	//  ╭───────────────────────────────────────────┬──────────╮
```
Comment (purplefox) [inline sst/sstable.go:35]: It might be worth saying this is the index.

**Comment by purplefox** [comment]
I should add: I am also going to write some architecture/design documentation that explains the database, and other stuff, and how they work, so that should help :)

File: sst/sstable.go (hunk 1 above)
Comment (naveen246) [reply sst/sstable.go:39]: Updated the comment mentioning the index

**Pull request merged**